src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlTokenStream.java

```diff
@@ -371,21 +371,23 @@ private final int _next() throws XMLStreamException
     
     private final String _collectUntilTag() throws XMLStreamException
     {
-        String text = null;
+        if (_xmlReader.isEmptyElement()) {
+            _xmlReader.next();
+            return null;
+        }
+
+        StringBuilder text = new StringBuilder();
```

Comment:
Hmmh. Why? This was an optimization for the common case of just single segment, in which case construction of intermediate `StringBuilder` was avoided. It's probably not a big deal either way, but usually I prefer not fixing things that ain't broke.